### PR TITLE
Feature/hide old wallet Dapp #2021

### LINKED ISSUE #2021: Wallet - hide old wallet Dapp

### PR DESCRIPTION
**Pull request by flexsurfer**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)
fixes #2021

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

### CODE REVIEW

**Review by yenda — APPROVED** [review]

**Review by janherich — APPROVED** [review]

File: src/status_im/chat/events.cljs [hunk 1]
```diff
@@ -180,11 +180,13 @@
         (let [chats            (->> all-stored-chats
                                     (map (fn [{:keys [chat-id] :as chat}]
                                            [chat-id (assoc chat :last-message (get-last-stored-message chat-id))]))
-                                    (into {}))]
+                                    (into {}))
+              ;;TODO temporary hide wallet chat, this code should be deleted after wallet contact will be deleted
+              chats' (dissoc chats "wallet")]
```
Comment (jeluard) [inline src/status_im/chat/events.cljs:185]: Can we came up with a better variable name?

File: src/status_im/ui/screens/contacts/events.cljs [hunk 2]
```diff
@@ -278,10 +278,12 @@
                                                         (assoc :bot id
                                                                :type :command))]))
                                (into {}))
-          contacts (into {} contacts-list)]
-      {:db         (assoc db :contacts/contacts contacts
+          contacts (into {} contacts-list)
+          ;;TODO temporary hide wallet contact, this code should be deleted after wallet contact will be deleted
+          contacts' (assoc-in contacts ["wallet" :pending?] true)]
```
Comment (jeluard) [inline src/status_im/ui/screens/contacts/events.cljs]: Same here, this variable name is not descriptive.

File: src/status_im/chat/events.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/chat/events.cljs:185]: tbh this code is too compact, you can see fast what happens here, and this is temporary, so i believe `temporary-chats-without-wallet` will be extra

File: src/status_im/ui/screens/contacts/events.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/contacts/events.cljs]: same here

**Comment by rasom** [comment]
tests are failing

File: src/status_im/chat/events.cljs (hunk 1 above)
Comment (oskarth) [reply src/status_im/chat/events.cljs:185]: @flexsurfer Just cursory look so not sure, but I think this dissoc might change conditions for test mocking.

Comment (flexsurfer) [reply src/status_im/chat/events.cljs:185]: fixed

Comment (yenda) [reply src/status_im/chat/events.cljs:185]: why use chat' in the first place ? you can you as-> and have everything in the chat in one threading macro.
also the map function could be moved into a function with a name, and reduce is better suited here than map then into {}

Comment (flexsurfer) [reply src/status_im/chat/events.cljs:185]: i don't want to incorporate this temporary workaround in the code somehow, it must be foreign and easily removable

**Comment by flexsurfer** [comment]
fixed tests and removed messages to wallet

**Comment by annadanchenko** [comment]
On 1st run (right after creating account): 
- go to Chats and see no Wallet (it's OK)
- go to Contacts and see Wallet in Contacts (it's not OK). Would expect that there is no Contact "Wallet"
https://app.testfairy.com/projects/4803590-status/builds/6822809/sessions/4/?accessToken=4myVLunuFNF6A3-AV/QvuaD2h9M
both iOS and Android

**Comment by flexsurfer** [comment]
@annadanchenko fixed

**Pull request merged**